### PR TITLE
Abbasc52/nested fix

### PR DESCRIPTION
**Pull request by abbasc52**
- Added GlobalParams support #97 
- LocalParams now work at all nested levels #98 
-  Added `Enabled` field to Rule to enable/disable a Rule #99 
- Fixed Rule compilation error not appearing as error message in certain cases #95

### CODE REVIEW

File: src/RulesEngine/RulesEngine.cs [hunk 1]
```diff
@@ -257,42 +257,22 @@ private bool RegisterRule(string workflowName, params RuleParameter[] ruleParams
...
+            {
+                throw new ArgumentException($"Workflow `{workflowName}` is not found");
+            }
+            var currentRule = workflow.Rules?.SingleOrDefault(c => c.RuleName == ruleName && c.Enabled);
             if (currentRule == null)
             {
                 throw new ArgumentException($"Workflow `{workflowName}` does not contain any rule named `{ruleName}`");
             }
-            return CompileRule(workflowName, ruleParameters, currentRule);
+            return CompileRule(currentRule, ruleParameters, workflow.GlobalParams?.ToArray());
```
Comment (abbasc52) [inline src/RulesEngine/RulesEngine.cs:270]: Check if scopedParams are disabled, are global params disabled as well

File: test/RulesEngine.UnitTest/ExpressionBuilders/RuleExpressionParserTests.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
+//using Moq;
```
Comment (abbasc52) [inline test/RulesEngine.UnitTest/ExpressionBuilders/RuleExpressionParserTests.cs:1]: Remove commented file

File: src/RulesEngine/RuleCompiler.cs [hunk 3]
```diff
@@ -2,12 +2,14 @@
 // Licensed under the MIT License.
```
Comment (abbasc52) [inline src/RulesEngine/RuleCompiler.cs:2]: check if CompiledParams are still getting populated in RuleResultTree

File: src/RulesEngine/RulesEngine.cs (hunk 1 above)
Comment (abbasc52) [reply src/RulesEngine/RulesEngine.cs:270]: Added test case for the same

File: src/RulesEngine/RuleCompiler.cs (hunk 3 above)
Comment (abbasc52) [reply src/RulesEngine/RuleCompiler.cs:2]: fixed and added test case for the same

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,14 @@
 
 All notable changes to this project will be documented in this file.
 
+## [3.1.0]
```
Comment (abbasc52) [inline CHANGELOG.md:5]: release as preview

**Review by dishantmunjal — APPROVED** [review]

**Pull request merged**